tests/chainerx_tests/unit_tests/routines_tests/test_creation.py

```diff
@@ -389,6 +389,16 @@ def test_ascontiguousarray_from_chainerx_array(device, shape, dtype, padding):
     assert e.dtype.name == a.dtype.name
 
 
+def test_ascontiguousarray_from_chainerx_array_device():
+    with chainerx.using_device(chainerx.get_device('native:0')):
+        dev = chainerx.get_device('native:1')  # Non default one
+        a = chainerx.arange(10, device=dev)
+        b = chainerx.ascontiguousarray(a)
+        assert b.is_contiguous
+        assert b.device == a.device
+        assert b.device != chainerx.get_default_device()
```

Comment:
Isn't this check redundant after the check in the previous line?

Reply:
Just in case I wanted to double check that I'm not using default device too.

Reply:
So what you really want to check is `dev is not chainerx.get_default_device()`?
Can you rewrite the check as such and move it to the line after `dev = ...`?
(or maybe we can just remove it as it's almost obvious.)

Reply:
So what you really want to check is `dev is not chainerx.get_default_device()`?
Can you rewrite the check as such and move it to the line after `dev = ...`?
(or maybe we can just remove it as it's almost obvious.)

Reply:
That's nice!